core/services/bulletprooftxmanager/eth_confirmer.go

```diff
@@ -225,6 +225,9 @@ func (ec *ethConfirmer) CheckForReceipts(ctx context.Context, blockNum int64) er
 	if len(attempts) == 0 {
 		return nil
 	}
+	if batchSize == 0 {
+		batchSize = len(attempts)
+	}
```

Comment:
is this mainly for tests? when do we expect it to be 0 ?

Reply:
It's just to protect ourselves in case someone sets it. If we don't special case 0, it will cause the application to enter an infinite loop.

Usually 0 means "disable" so that's what I added here

Reply:
ok 👍🏼